app/models/chapter.rb

```diff
@@ -90,6 +90,29 @@ def fix_positions_after_destroy
     end
   end
 
+  after_commit :update_series_index
+  def update_series_index
+    return unless work&.series.present? && should_reindex_series?
+    work.series.each do |series|
+      # ES UPGRADE TRANSITION #
```

Comment:
IndexQueue checks rollout to see which ES version to use, so you don't need to check that here, you can just do:

```rb
IndexQueue.enqueue_ids(Series, work.series.pluck(:id), :main)
work.series.each do |series|
  IndexQueue.enqueue_ids(Bookmark, series.bookmarks.pluck(:id), :background)
end
```

Reply:
Oh, nice, thanks!

It's _technically_ outside the scope of this, but I think I'll change it in https://github.com/otwcode/otwarchive/blob/8a28b63145973fa40701f027931872896765641a/app/models/serial_work.rb#L28 as well, since that's how I ended up with this >.>